src/ringct/rctTypes.h

```diff
@@ -535,14 +535,14 @@ void b2h(key &amountdh, bits amountb2);
 //int[64] to uint long long
 xmr_amount b2d(bits amountb);
 
-static inline const rct::key pk2rct(const crypto::public_key &pk) { return (const rct::key &)pk; }
-static inline const rct::key sk2rct(const crypto::secret_key &sk) { return (const rct::key &)sk; }
-static inline const rct::key ki2rct(const crypto::key_image &ki) { return (const rct::key &)ki; }
-static inline const rct::key hash2rct(const crypto::hash &h) { return (const rct::key &)h; }
-static inline const crypto::public_key rct2pk(const rct::key &k) { return (const crypto::public_key &)k; }
-static inline const crypto::secret_key rct2sk(const rct::key &k) { return (const crypto::secret_key &)k; }
-static inline const crypto::key_image rct2ki(const rct::key &k) { return (const crypto::key_image &)k; }
-static inline const crypto::hash rct2hash(const rct::key &k) { return (const crypto::hash &)k; }
+static inline const rct::key& pk2rct(const crypto::public_key &pk) { return (const rct::key&)pk; }
```

Comment:
After thinking again about this part of the code I think the best way is to rename this "casts" to `cast2x` or `reinterprete2x` where `x` is the name of the result. It is also possible to use a template `cast<X>()`  but this makes the caller code much longer.

Reply:
I also had a think about it, and it isn't worth doing a critical bug over. In the near future I will bring over a generic `vector32` class that can be typedef'ed to all of those (additional benefit is that then the keys are wiped consistently by the destructor instead of piecemeal like they are now)

Reply:
Oh just FIY the lines still changed because I updated the definition from "kool kid" style `type *var` to more sensible `type* var`.